### PR TITLE
Feat/webusb

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/CHI-4522

Reference for webusb implementation: https://developer.chrome.com/articles/build-for-webusb/
Reference for HID report: https://ravikiranb.com/tutorials/generic-comm-usb-hid/

### CODE REVIEW

File: libusb/inc/usb_cdc.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
  * @{ */
 #define USB_CLASS_CDC                   0x02    /**<\brief Communicational Device class */
 #define USB_CLASS_CDC_DATA              0x0A    /**<\brief Data Interface class */
+#define USB_CLASS_VENDOR                0xFF
```
Comment (ujjwal-cyph) [inline libusb/inc/usb_cdc.h]: probably these changes are not needed. please take a look

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4522 WebUSB Firmware Enabler](https://app.clickup.com/t/9002019994/CHI-4522)

File: libusb/libusb.c [hunk 2]
```diff
@@ -19,17 +19,40 @@
 #include "stm32.h"
 #include "usb.h"
 #include "usb_cdc.h"
+#include "usb_hid.h"
```
Comment (kartikCypherock) [inline libusb/libusb.c:23]: If we can wrap these includes using `#if (ENABLE_CDC_COMM == 1)` / `#elif (ENABLE_HID_WEBUSB_COMM == 1)`, then we can be sure that CDC communication will not be impacted which is explicitly used by the bootloader

File: libusb/libusb.c [hunk 3]
```diff
@@ -84,18 +126,46 @@ static const struct usb_device_descriptor device_desc = {
...
+#elif (ENABLE_HID_WEBUSB_COMM == 1)
         .bNumInterfaces         = 2,
+#else
+    #error "No USB protocol defined select ENABLE_CDC_COMM or ENABLE_HID_WEBUSB_COMM"
+#endif
         .bConfigurationValue    = 1,
         .iConfiguration         = NO_DESCRIPTOR,
         .bmAttributes           = USB_CFG_ATTR_RESERVED | USB_CFG_ATTR_SELFPOWERED,
-        .bMaxPower              = USB_CFG_POWER_MA(100),
+        .bMaxPower              = USB_CFG_POWER_MA(200),
```
Comment (ujjwal-cyph) [inline libusb/libusb.c:168]: just take notice that is intended

File: libusb/libusb.c [hunk 4]
```diff
@@ -252,60 +507,220 @@ static usbd_respond cdc_control(usbd_device *dev, usbd_ctlreq *req, usbd_rqc_cal
...
     }
 }
 
-void cdc_init_usbd(void) {
-    usbd_init(&libusb_udev, &usbd_hw, CDC_EP0_SIZE, ubuf, sizeof(ubuf));
-    usbd_reg_config(&libusb_udev, cdc_setconf);
-    usbd_reg_control(&libusb_udev, cdc_control);
-    usbd_reg_descr(&libusb_udev, cdc_getdesc);
+void usb_device_config_init(void) {
+    usbd_init(&libusb_udev, &usbd_hw, 8, ubuf, sizeof(ubuf));
```
Comment (ujjwal-cyph) [inline libusb/libusb.c]: here macro is replaced with a constant value. make sure it is expected

File: libusb/inc/usb_cdc.h (hunk 1 above)
Comment (amanCypherock) [reply libusb/inc/usb_cdc.h]: removed

File: libusb/libusb.c (hunk 3 above)
Comment (amanCypherock) [reply libusb/libusb.c:168]: yeaaaaaaaaaaaaaaah

File: libusb/libusb.c (hunk 2 above)
Comment (amanCypherock) [reply libusb/libusb.c:23]: updated

File: libusb/libusb.c (hunk 4 above)
Comment (amanCypherock) [reply libusb/libusb.c]: updated

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**